### PR TITLE
feat: optional connection in trigger

### PR DESCRIPTION
**Pull request by ig-ra**
allow connection to be optional in trigger

### CODE REVIEW

File: internal/backend/db/dbgorm/triggers.go [hunk 1]
```diff
@@ -14,24 +14,33 @@ import (
 
 func triggerToRecord(ctx context.Context, tx *tx, trigger sdktypes.Trigger) (*scheme.Trigger, error) {
 	connID := trigger.ConnectionID()
+	var conn sdktypes.Connection
+	var err error
+	var projID sdktypes.ProjectID
```
Comment (efiShtain) [inline internal/backend/db/dbgorm/triggers.go]: nit
this is more "go" way
```suggestion
	var (
	       conn sdktypes.Connection
	       err error
	       projID sdktypes.ProjectID
	  )
```

File: internal/backend/db/dbgorm/triggers.go [hunk 2]
```diff
@@ -14,24 +14,33 @@ import (
...
 		}
 
-		if projID != env.ProjectID() {
+		if projID.IsValid() && projID != env.ProjectID() {
 			return nil, fmt.Errorf("env and connection project mismatch: %v != %v", projID, env.ProjectID())
 		}
+		projID = env.ProjectID()
+	}
+
+	if !projID.IsValid() {
```
Comment (efiShtain) [inline internal/backend/db/dbgorm/triggers.go:44]: move this check after line 26 and fail fast

then you can remove the projID.IsValid() in line 36 as well

File: proto/autokitteh/triggers/v1/trigger.proto [hunk 3]
```diff
@@ -9,7 +9,7 @@ import "buf/validate/validate.proto";
 message Trigger {
   string trigger_id = 1;
   string name = 2;
-  string connection_id = 3;
+  string connection_id = 3; // could be empty
```
Comment (efiShtain) [inline proto/autokitteh/triggers/v1/trigger.proto:12]: you don't really this comment
all fields are optional in proto by default

File: sdk/sdktypes/trigger.go [hunk 4]
```diff
@@ -89,11 +88,20 @@ func (p Trigger) ToValues() map[string]Value {
 	return map[string]Value{
 		"name": NewStringValue(p.read().Name),
 		"data": kittehs.Must1(NewStructValue(
-			NewStringValue("event_data"),
+			NewStringValue("trigger_data"),
```
Comment (efiShtain) [inline sdk/sdktypes/trigger.go:99]: not sure I understand what this is doing
should this be a constant somewhere ?

**Review by efiShtain — APPROVED** [review]

File: sdk/sdktypes/trigger.go (hunk 4 above)
Comment (ig-ra) [reply sdk/sdktypes/trigger.go:99]: it's just fake name. Probably cut-n-pasted from event code. So I changed the existing typo in code.

File: proto/autokitteh/triggers/v1/trigger.proto (hunk 3 above)
Comment (ig-ra) [reply proto/autokitteh/triggers/v1/trigger.proto:12]: you are right. But I want to provide addition info why there is no validation check

**Review by efiShtain — APPROVED** [review]

**Pull request merged**